### ISSUE TITLE
Nightly Tests of StructuralMechanics take too long

### ISSUE THREAD

**Issue by philbucher** [KratosMultiphysics/Kratos#1016]
```
Running tests for StructuralMechanicsApplication
.........................................................................s....
ABORT: Tests for StructuralMechanicsApplication took to long. Process Killed.
```
@KratosMultiphysics/structural-mechanics please only add short running tests, we are already taking too long.

@roigcarlo is it possible to increase the time? What is the current limit? On my machine the tests take ~30 seconds.
I know which tests are taking too long and the are on my list of getting refactored, but not soon.
Also there will be more tests / more things to test in the future, so the time might even increase a bit more

What do you suggest as solution?

**Comment by roigcarlo** [comment]
Hi @philbucher 
Sorry for taking that long to answer but I wanted to make sure so I've run some tests of my own. 

Test should take like 10 min, but the hard-kill you see triggers at 15 min.

We are already far beyond the limit. In fact you may have notice that every now and then clang tests stop running, that means that we have reached the limit again so we have to re-extend.

The problem are not the tests per-se but that Kratos is compiled with -O0 in amazon. This reduces the compile time of kratos but it increases the runtime of the some tests by around x10. Actually I've tested the runtimes with both options and there is only a subset of tests that are affected with -0X options. I suspect those are the ones that run small problems to check some results. Initially this was not an issue because we had maybe one or two of those tests, but now there are more, and not only in the StructuralMechanics app.

Nevertheless this is unfair and I know it. When we set up the nightly builds that was the only thing that gave us reasonable times, which is clearly not the case anymore. 

In conclusion, we have two problems:
- We are compiling with -O0.
- There are unittests that are result validations rather than unittests.

My proposal would be this: I will take a look at the builds and try to squish more the resources of the machine, but it also would be nice if you could break some of the longer tests in small ones.

**Comment by jcotela** [comment]
Just to add to what @roigcarlo is saying, I think we should consider moving all "small problem" tests to one of the low-priority test suites, and just run real unit tests in the nightly builds.

We have the same problem in the fluid dynamics application, and our plan is to move to very trivial tests (build an elemental matrix for a given formulation, check that it still gives the same numbers as last time we validated that formulation) and move full examples (even if small) to the validation suite.

**Comment by loumalouomega** [comment]
I agree to move most of the test to validation and just keep the short ones as nightly

**Comment by philbucher** [comment]
Thx a lot for you help!
@roigcarlo are the nightly tests not running on a CIMNE-machine?

I will for now move the slow tests to the validation suite then. Just to confirm, the tests that are being executed are the `nightly` ones?

**Comment by roigcarlo** [comment]
Tests being executed are Small and nightly.

And no, tests run on the m4.large tier of aws: https://aws.amazon.com/es/ec2/instance-types/ for more info

**Comment by philbucher** [comment]
Ah I didn't know that.

[So I should not add the smalltests to the nightsuite then](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/tests/test_StructuralMechanicsApplication.py#L241), otherwise they are run twice?

**Comment by loumalouomega** [comment]
No, but we should move many nightly to validation

**Comment by philbucher** [comment]
ok then I know what to do, thx guys

**Comment by roigcarlo** [comment]
Yep sorry my bad, I run "nightly" and I assume small are there as well, you are doing right :P

**Comment by philbucher** [comment]
ok thx
